api/pkg/service/catalog/catalog.go

```diff
@@ -57,24 +57,28 @@ func (s *service) Refresh(ctx context.Context, p *catalog.RefreshPayload) (*cata
 	log := s.Logger(ctx)
 	db := s.DB(ctx)
 
-	log.Infof("going to enqueue")
-
 	ctg := model.Catalog{}
-	if err := db.Model(&model.Catalog{}).First(&ctg).Error; err != nil {
+	if err := db.Where(&model.Catalog{Name: p.CatalogName}).Model(&model.Catalog{}).First(&ctg).Error; err != nil {
 		if err == gorm.ErrRecordNotFound {
-			return nil, notFoundError
+			return nil, catalogNotFoundErr(p.CatalogName)
 		}
 		log.Error(err)
 		return nil, internalError
 	}
 
+	log.Infof("going to enqueue")
+
 	job, err := s.wq.Enqueue(auth.UserID(ctx), ctg.ID)
 	if err != nil {
 		return nil, err
 	}
 
-	ret := &catalog.Job{ID: job.ID, Status: job.Status}
+	ret := &catalog.Job{ID: job.ID, CatalogName: ctg.Name, Status: job.Status}
 	log.Infof("job %d queued for refresh", job.ID)
 
 	return ret, nil
 }
+
+func catalogNotFoundErr(name string) error {
```

Comment:
Why are we not adding this error in goa design and tweaking the error based on it ??